protos/feast/core/OnDemandFeatureView.proto

```diff
@@ -49,9 +50,11 @@ message OnDemandFeatureViewSpec {
     map<string, OnDemandSource> sources = 4;
 
     oneof transformation {
```

Comment:
This hasn't made it to an official release yet, so can't we remove `transformation` oneof here? just leave `UserDefinedFunction user_defined_function = 5 [deprecated = true]`. It would also enable you to call the new field `transformation` which sounds less awkward (`FeatureTransformationV2 transformation = 10;`) and cut down on overall code size.

Reply:
I'm going to do that in a separate PR as a follow up. I'll probably end up with 4 by the end. I want to leave this one as is though.

Reply:
It's better to have separate clean PRs than one massive one imo.

Reply:
@tokoko lmk if you're good with this. I have the next PR coming soon where I'll rename things and the subsequent will do the PythonTransformation type.

Reply:
@franciscojavierarceo yeah, sounds good to me. Let's just not make a new release until we get all these ironed out.